app/queries/users_by_thread_volume_query.rb

```diff
@@ -0,0 +1,22 @@
+class UsersByThreadVolumeQuery
+  def self.email(discussion)
+    volume_and_discussion(:email, discussion)
+  end
+
+  def self.normal(discussion)
+    volume_and_discussion(:normal, discussion)
+  end
+
+  def self.mute(discussion)
+    volume_and_discussion(:mute, discussion)
+  end
+
+  private
+  def self.volume_and_discussion(volume, discussion)
+    User.
+      active.
+      joins("LEFT OUTER JOIN discussion_readers dr ON (dr.user_id = users.id AND dr.discussion_id = #{discussion.id})").
```

Comment:
Probably fine, but just to be more safest, `... dr.discussion_id = ?", discussion.id`